### PR TITLE
Fix #287 - Document metrics available by source

### LINKED ISSUE #287: consider documenting which metrics are available by source type

### PR DESCRIPTION
**Pull request by parmsam-pfizer**
- Create vignette to start capturing this in documentation. PR associated with #287.

### CODE REVIEW

**Comment by parmsam-pfizer** [comment]
@emilliman5 
Can you share the code you used to generate the generic calls by dispatch source in [this video](https://youtu.be/EqKhmJ1lcN8?si=41KJ9pn2w8bBmjTR&t=865)? Did you use the code you already shared in #287?

**Comment by emilliman5** [comment]
@AARON-CLARK, should we create a function to generate a table of available assessments and metrics by ref source or would a rendered vignette be sufficient?

**Comment by emilliman5** [comment]
@parmsam-pfizer here is the code to create a rudimentary table linking assessments to ref source.

ns <- getNamespaceExports("riskmetric")

    ns <- grep("^pkg_ref_cache|^assess_", ns, value = T, 
               ignore.case = T)
    ns <- sapply(ns, methods)

    ns_df <- tibble(export=unlist(ns)) %>%
      bind_rows(tibble(export=names(ns)[sapply(ns, length)==0])) %>% 
      separate(export, c("generic", "cache", "class"), 
               sep = "\\.", remove = F)

    ns_df %>% filter(grepl("assess", generic)) %>%
      select(generic, class=cache) %>%
      mutate(value=1) %>%
      pivot_wider(names_from = class, values_from = value) %>% 
      mutate(default = coalesce(default, `NA`)) %>% 
      select(generic, default, everything(), -`NA`)

**Comment by parmsam** [comment]
Pasting this reference code from personal Github for a later commit: 
```{r}
library(riskmetric)
library(dplyr)
library(tibble)
library(tidyr)
library(gt)
library(gtExtras)

ns <- getNamespaceExports("riskmetric")

ns <- grep("^pkg_ref_cache|^assess_", ns, value = T, 
           ignore.case = T)
ns <- sapply(ns, methods)

ns_df <- tibble(export=unlist(ns)) %>%
  bind_rows(tibble(export=names(ns)[sapply(ns, length)==0])) %>% 
  separate(export, c("generic", "cache", "class"), 
           sep = "\\.", remove = F)

cross_tbl <- ns_df %>% filter(grepl("assess", generic)) %>%
  select(generic, class=cache) %>%
  mutate(value=1) %>%
  pivot_wider(names_from = class, values_from = value) %>% 
  mutate(default = coalesce(default, `NA`)) %>% 
  select(generic, default, everything(), -`NA`)

cross_tbl %>% 
  dplyr::mutate(
    across(c(everything(), -generic), 
           function(x) ifelse(is.na(x), "x", "check"))
  ) %>%
  gt() %>%
  gt_fa_column(
    column = c(everything(), -generic),
    palette = c(
      "check" = "#009E73", 
      "x" = "#D55E00"
    )
  )
```

**Review by emilliman5 — CHANGES_REQUESTED** [review]
Please add a little bit for text describing each table. Maybe add a title to the tables as well. Does not need to be extensive but a little context will help users understand what the tables mean.

File: vignettes/available-metrics.Rmd [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup echo = FALSE}
```
Comment (emilliman5) [inline vignettes/available-metrics.Rmd:17]: need to add a "," between chunk name and parameters

**Review by emilliman5 — APPROVED** [review]

**Pull request merged**